app/components/homeScreen/DailyReads.tsx

```diff
@@ -228,52 +245,88 @@ const DailyReads = ():any => {
+      console.log('First time here', categoryActivityData.length);
+      //const activityDataToShow = categoryActivityData[Math.floor(Math.random() * categoryActivityData.length)];
+      const indexNew = Math.floor(Math.random() * categoryActivityData.length);
+      const activityDataToShow = categoryActivityData[indexNew];
+      categoryActivityData.splice(indexNew, 1);
+
+      const indexNew2 = Math.floor(Math.random() * categoryActivityData.length);
+      const activityDataToShow1 = categoryActivityData[indexNew2];
+      const articleListData: any = [];
+      const activityListData: any = [];
       if (articleDataToShow != null && articleDataToShow != undefined) {
         advicearray.push(articleDataToShow?.id);
-        dailyArticles.push(articleDataToShow);
+        advicearray.push(articleDataToShow1?.id);
+        articleListData.push(articleDataToShow);
+        articleListData.push(articleDataToShow1);
       }
       if (activityDataToShow != null && activityDataToShow != undefined) {
         gamesarray.push(activityDataToShow?.id);
-        dailyActivityList.push(activityDataToShow);
+        gamesarray.push(activityDataToShow1?.id);
+        activityListData.push(activityDataToShow);
+        activityListData.push(activityDataToShow1);
       }
-      setDataToShowInList(dailyArticles);
-      setActivityDataToShowInList(dailyActivityList);
+      setDataToShowInList(articleListData);
+      setActivityDataToShowInList(activityListData);
       const dailyDataCategorytoDispatch: any = { ...dailyDataCategoryall };
       const showedDailyDataCategorytoDispatch: any = { ...showedDailyDataCategoryall };
       dailyDataCategorytoDispatch[activeChild.uuid] = {
-        advice: articleDataToShow && articleDataToShow != null ? articleCategoryArrayNew[nextIndex] : 0,
-        games: activityDataToShow && activityDataToShow != null ? activityCategoryArrayNew[nextIndex2]?.id : 0,
-        currentadviceid: articleDataToShow && articleDataToShow != null ? articleDataToShow?.id : 0,
-        currentgamesid: activityDataToShow && activityDataToShow != null ? activityDataToShow?.id : 0,
+        advice: [articleDataToShow && articleDataToShow != null ? articleCategoryArrayNew[nextIndex] : 0,
+        articleDataToShow1 && articleDataToShow1 != null ? articleCategoryArrayNew[nextIndex + 1] : 0],
+        games: [activityDataToShow && activityDataToShow != null ? activityCategoryArrayNew[nextIndex2]?.id : 0,
+        activityDataToShow1 && activityDataToShow1 != null ? activityCategoryArrayNew[nextIndex2 + 1]?.id : 0
+        ],
+        currentadviceid: [articleDataToShow && articleDataToShow != null ? articleDataToShow?.id : 0,
+        articleDataToShow1 && articleDataToShow1 != null ? articleDataToShow1?.id : 0],
+        currentgamesid: [activityDataToShow && activityDataToShow != null ? activityDataToShow?.id : 0,
+        activityDataToShow1 && activityDataToShow1 != null ? activityDataToShow1?.id : 0

```

Comment:
`articleDataToShow1?.id`

? mark is not necessary since you already check for null.